### PR TITLE
BUG: Use args and kwargs in Rolling.apply

### LINKED ISSUE #33433: BUG: Using `kwargs` in `Rolling.apply()`
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of pandas.

- [ ] (optional) I have confirmed this bug exists on the master branch of pandas.

---

#### Code Sample, a copy-pastable example

```python
from functools import partial

import numpy as np
import pandas as pd


def foo(x, par):
    return np.sum(x + par)


df = pd.DataFrame({"gr": [1, 1], "a": [1, 2]})

# With `kwargs`
try:
    print(df.groupby("gr")["a"].rolling(1).apply(foo, kwargs={"par": 10}))
except Exception as e:
    print(e)

# Using `partial`
bar = partial(foo, par=10)
print(df.groupby("gr")["a"].rolling(1).apply(bar))
```

#### Problem description

**Expected**: using appropriate `kwargs` argument for `pandas.core.window.rolling.Rolling.apply()` should be equivalent to premodifying supplied function with `functools.partial()` (because it is stated in [documentation](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.core.window.rolling.Rolling.apply.html)).
**Current**: `kwargs` doesn't seem to be used at all, because it results into "missing 1 required positional argument" exception.
**Note**: This works as expected in pandas 0.25.3, but gives error as early as 1.0.0.

#### Expected Output

```
gr   
1   0    11.0
    1    12.0
Name: a, dtype: float64
gr   
1   0    11.0
    1    12.0
Name: a, dtype: float64
```

#### Output of ``pd.show_versions()``

<details>

INSTALLED VERSIONS
------------------
commit           : None
python           : 3.8.2.final.0
python-bits      : 64
OS               : Linux
OS-release       : 4.15.0-29-generic
machine          : x86_64
processor        : x86_64
byteorder        : little
LC_ALL           : None
LANG             : ru_UA.UTF-8
LOCALE           : ru_UA.UTF-8

pandas           : 1.0.3
numpy            : 1.18.1
pytz             : 2019.3
dateutil         : 2.8.1
pip              : 20.0.2
setuptools       : 46.1.3.post20200325
Cython           : None
pytest           : None
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : None
jinja2           : None
IPython          : None
pandas_datareader: None
bs4              : None
bottleneck       : None
fastparquet      : None
gcsfs            : None
lxml.etree       : None
matplotlib       : None
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pytables         : None
pytest           : None
pyxlsb           : None
s3fs             : None
scipy            : None
sqlalchemy       : None
tables           : None
tabulate         : None
xarray           : None
xlrd             : None
xlwt             : None
xlsxwriter       : None
numba            : None

</details>

### PR DESCRIPTION
**Pull request by mproszewska**
- [x] closes #33433
- [x] tests added / passed
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/tests/window/test_apply.py [hunk 1]
```diff
@@ -138,3 +138,13 @@ def test_invalid_kwargs_nopython():
         Series(range(1)).rolling(1).apply(
             lambda x: x, kwargs={"a": 1}, engine="numba", raw=True
         )
+
+
+def test_apply_kwargs():
+    def foo(x, par):
```
Comment (mroeschke) [inline pandas/tests/window/test_apply.py:144]: Could you add the Github issue number as a comment? `# GH ...`
Could you also test passing `args`?

File: pandas/tests/window/test_apply.py [hunk 2]
```diff
@@ -138,3 +138,13 @@ def test_invalid_kwargs_nopython():
...
+        return np.sum(x + par)
+
+    df = DataFrame({"gr": [1, 1], "a": [1, 2]})
+    result = df.groupby("gr")["a"].rolling(1).apply(foo, kwargs={"par": 10})
+    expected = np.array([11.0, 12.0])
+    tm.assert_numpy_array_equal(result.values, expected)
```
Comment (mroeschke) [inline pandas/tests/window/test_apply.py:150]: Could you use `tm.assert_frame/series_equals` instead?

**Review by jreback — CHANGES_REQUESTED** [review]
needs a release note.

we actually want to test this for both .rolling.apply as well as .groupby.rolling.apply (as you did).

File: pandas/tests/window/test_apply.py (hunk 1 above)
Comment (simonjayhawkins) [reply pandas/tests/window/test_apply.py:144]: > Could you also test passing `args`?

+1

**Review by simonjayhawkins — CHANGES_REQUESTED** [review]

**Review by jreback — CHANGES_REQUESTED** [review]
pls also add a whatsnew note, 1.1, bug fixes, in groupby/rollining section

File: doc/source/whatsnew/v1.1.0.rst [hunk 3]
```diff
@@ -610,6 +610,11 @@ Groupby/resample/rolling
 - Bug in :meth:`DataFrame.groupby` where a ``ValueError`` would be raised when grouping by a categorical column with read-only categories and ``sort=False`` (:issue:`33410`)
 - Bug in :meth:`GroupBy.first` and :meth:`GroupBy.last` where None is not preserved in object dtype (:issue:`32800`)
 
+Groupby/rolling
+^^^^^^^^^^^^^^^^^^^^^^^^
+
+- Bug in :meth:`GroupBy.rolling.apply` ignores args and kwargs parameters (:issue:`33433`)
```
Comment (mroeschke) [inline doc/source/whatsnew/v1.1.0.rst:616]: You can put this in the section above `Groupby/resample/rolling`

File: pandas/tests/window/test_apply.py [hunk 4]
```diff
@@ -138,3 +138,31 @@ def test_invalid_kwargs_nopython():
...
+    # GH 33433
+    def foo(x, par):
+        return np.sum(x + par)
+
+    df = DataFrame({"gr": [1, 1], "a": [1, 2]})
+
+    idx = Index(["gr", "a"])
+    expected = DataFrame([[11.0, 11.0], [11.0, 12.0]], columns=idx)
+
+    result = df.rolling(1).apply(foo, kwargs={"par": 10})
```
Comment (mroeschke) [inline pandas/tests/window/test_apply.py:153]: Could you parameterize over the kwargs & args?

`pytest.mark.parameterize('args_kwargs', [[None, {"par": 10}], [(10,), None]])`

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
thanks @mproszewska 

cc @simonjayhawkins marked for 1.04 as well

**Pull request merged**